src/xrCore/Animation/Bone.hpp

```diff
@@ -465,41 +455,43 @@ class XRCORE_API CBoneData : public IBoneData
     using ChildFacesVec = xr_vector<FacesVec>;
     ChildFacesVec child_faces; // shared
 
-    CBoneData(u16 ID) : SelfID(ID) { VERIFY(SelfID != BI_NONE); }
-    virtual ~CBoneData() {}
+    explicit CBoneData(u16 ID) : SelfID(ID) { VERIFY(SelfID != BI_NONE); }
+    virtual ~CBoneData() = default;
 #ifdef DEBUG
     typedef svector<int, 128> BoneDebug;
     void DebugQuery(BoneDebug& L);
 #endif
     IC void SetParentID(u16 id) { ParentID = id; }
-    IC u16 GetSelfID() const { return SelfID; }
-    IC u16 GetParentID() const { return ParentID; }
+    [[nodiscard]] IC u16 GetSelfID() const override { return SelfID; }
+    [[nodiscard]] IC u16 GetParentID() const override { return ParentID; }
     // assign face
     void AppendFace(u16 child_idx, u16 idx) { child_faces[child_idx].push_back(idx); }
     // Calculation
     void CalculateM2B(const Fmatrix& Parent);
 
 private:
-    IBoneData& GetChild(u16 id) { return *children[id]; }
-    const IBoneData& GetChild(u16 id) const { return *children[id]; }
-    u16 GetNumChildren() const { return (u16)children.size(); }
-    const SJointIKData& get_IK_data() const { return IK_data; }
-    const Fmatrix& get_bind_transform() const { return bind_transform; }
-    const SBoneShape& get_shape() const { return shape; }
-    const Fobb& get_obb() const { return obb; }
-    const Fvector& get_center_of_mass() const { return center_of_mass; }
-    float get_mass() const { return mass; }
-    virtual u16 get_game_mtl_idx() const override { return game_mtl_idx; }
-    virtual shared_str GetMaterialName() const override { return name; }
-    float lo_limit(u8 k) const { return IK_data.limits[k].limit.x; }
-    float hi_limit(u8 k) const { return IK_data.limits[k].limit.y; }
+    IBoneData& GetChild(u16 id) override { return *children[id]; }
+    [[nodiscard]] const IBoneData& GetChild(u16 id) const override { return *children[id]; }
+    [[nodiscard]] u16 GetNumChildren() const override { return children.size(); }
```

Comment:
```suggestion
    [[nodiscard]] u16 GetNumChildren() const override { return (u16)children.size(); } // XXX: maybe change return type to size_t instead of u16. Check later.
```

Reply:
Added.